crates/forge/bin/cmd/test/mod.rs

```diff
@@ -662,9 +666,25 @@ impl TestArgs {
 
             // Write gas snapshots to disk if any were collected.
             if !gas_snapshots.is_empty() {
-                // Check for differences in gas snapshots if `FORGE_SNAPSHOT_CHECK` is set.
+                // By default `gas_snapshot_check` is set to `false` in the config.
+                //
+                // The user can either:
+                // - Set `FORGE_SNAPSHOT_CHECK=true` in the environment.
+                // - Pass `--gas-snapshot-check=true` as a CLI argument.
+                // - Set `gas_snapshot_check = true` in the config.
+                //
+                // If the user passes `--gas-snapshot-check=<bool>` then it will override the config
+                // and the environment variable, disabling the check if `false` is passed.
+                //
                 // Exiting early with code 1 if differences are found.
-                if std::env::var("FORGE_SNAPSHOT_CHECK").is_ok() {
+                let should_check_for_differences =
```

Comment:
nit `if self.gas_snapshot_check.unwrap_or_else(|| config.gas_snapshot_check) {`

Reply:
fixed in equivalent https://github.com/foundry-rs/foundry/pull/9791/commits/d56e16d7d340972c8ee48db43803ef3a336c2b11